### ISSUE TITLE
inconsistent sentence parsing based on whitespace after sentence terminal punctuation

### ISSUE THREAD

**Issue by timbowhite** [wooorm/parse-english#12]
Hi.  Thanks for this module, one of the better NLP's out there.

It appears sentences are parsed differently if there is no whitespace following the ending punctuation of the sentence.  Of course, I think the lack of at least 1 whitespace separating sentences is syntactically incorrect English.

Here's examples of 2 sentences with various ending punctuation, some with separating whitespace, others without:
```
var English = require('parse-english'),
    toString = require('nlcst-to-string');
    tests = [
        'This module works quite well. I also like to eat cheese.',
        'This module works quite well.I also like to eat cheese.',
        'This module works quite well?I also like to eat cheese.',
        'This module works quite well? I also like to eat cheese.',
        'This module works quite well!I also like to eat cheese.',
        'This module works quite well! I also like to eat cheese.',
        '"This module works quite well."I also like to eat cheese.',
        '"This module works quite well." I also like to eat cheese.',
        '"This module works quite well".I also like to eat cheese.',
        '"This module works quite well". I also like to eat cheese.',
        '\'This module works quite well\'.I also like to eat cheese.',
        '\'This module works quite well\'. I also like to eat cheese.',
        'This module works (quite well).I also like to eat cheese.',
        'This module works (quite well). I also like to eat cheese.',
    ];


tests.forEach(function(text){
    var tree = new English().parse(text),
        sentences = [];

    tree.children.forEach(function(paragraph){
        paragraph.children.forEach(function(sentence){
            var s;
            if (sentence.type !== 'SentenceNode' || ! (s = toString(sentence).trim())) return;
            sentences.push(s);
        });
    });

    console.log(sentences, sentences.length);
});
```

Output:
```
[ 'This module works quite well.',
  'I also like to eat cheese.' ] 2
[ 'This module works quite well.I also like to eat cheese.' ] 1
[ 'This module works quite well?I also like to eat cheese.' ] 1
[ 'This module works quite well?',
  'I also like to eat cheese.' ] 2
[ 'This module works quite well!',
  'I also like to eat cheese.' ] 2
[ 'This module works quite well!',
  'I also like to eat cheese.' ] 2
[ '"This module works quite well."',
  'I also like to eat cheese.' ] 2
[ '"This module works quite well."',
  'I also like to eat cheese.' ] 2
[ '"This module works quite well".',
  'I also like to eat cheese.' ] 2
[ '"This module works quite well".',
  'I also like to eat cheese.' ] 2
[ '\'This module works quite well\'.I also like to eat cheese.' ] 1
[ '\'This module works quite well\'.',
  'I also like to eat cheese.' ] 2
[ 'This module works (quite well).',
  'I also like to eat cheese.' ] 2
[ 'This module works (quite well).',
  'I also like to eat cheese.' ] 2
```

So in summary, ```parse-english``` doesn't separate sentences that end in either ```.``` or ```?``` with no separating whitespace.  Is this expected behavior?

**Comment by wooorm** [comment]
Thanks!

This is expected behaviour, as white-space is an integral part of Latin / English language.
Sentences are pretty hard to detect, there’s terminal markers (`!`, `?`, `.`) in many other places too,
so the white-space is (currently?) required. However, if you know an alternative way to distinguish sentences, I’m all ears!

**Comment by timbowhite** [comment]
> However, if you know an alternative way to distinguish sentences, I’m all ears!

Besides checking non-spaced sentence terminations against common abbreviations (which this module already appears to do, among other things), I don't at the moment.

Feel free to close :)

**Comment by wooorm** [comment]
> Besides checking non-spaced sentence terminations against common abbreviations (which this module already appears to do, among other things), I don't at the moment.

The abbreviations are used to classify some things as non-sentences: `Mr. McFoo` or something. I feel using that list, which isn’t exhaustive, to do the inverse causes more problems than it solves!

Closing, let me know if you have more Qs!
